libraries/BluetoothSerial/src/BluetoothSerial.cpp

```diff
@@ -384,6 +384,22 @@ bool BluetoothSerial::begin(String localName)
     return _init_bt(local_name.c_str());
 }
 
+bool BluetoothSerial::pinCode( char* pswd)
+{
+    if(strlen(pswd)==4){
+        esp_bt_pin_type_t pin_type = ESP_BT_PIN_TYPE_FIXED;
+        esp_bt_pin_code_t pin_code;
+        pin_code[0] = pswd[0];
+        pin_code[1] = pswd[1] ;
+        pin_code[2] = pswd[2];
+        pin_code[3] = pswd[3];
+        esp_bt_gap_set_pin(pin_type, 4, pin_code);
+        return true;
+    }
+    else
```

Comment:
is the limit 4 chars?

Reply:
```cpp
#define ESP_BT_PIN_CODE_LEN        16                   /*!< Max pin code length */
typedef uint8_t esp_bt_pin_code_t[ESP_BT_PIN_CODE_LEN]; /*!< Pin Code (upto 128 bits) MSB is 0 */
```

What is the reason for the limitaton?

Reply:
@me-no-dev 
Sorry for the late reply, I will modify the code, rebase and submit again.